### PR TITLE
Fix unhelpful error in top-level await malformed expression.

### PR DESCRIPTION
**Pull request by nchevobbe**
In mapExpression, if we fail to get an AST
from the expression this means the expression
has a syntax error in it.
In order to get a meaningful error when evaluating
top-level await expression, we still need to
wrap the expression in an async iife.
This required some changes in mapExpression in order
to not bail completely if parsing the expression
threw.
We also take this as an opportunity to only parse
the expression once for the all the transforms.

### CODE REVIEW

File: src/workers/parser/mapExpression.js [hunk 1]
```diff
@@ -30,22 +31,36 @@ export default function mapExpression(
...
+      return ast;
+    }
+
+    try {
+      ast = parseConsoleScript(expression);
+    } catch (e) {
+      // no need to do anything
+    }
+    return ast;
+  };
```
Comment (jasonLaster) [inline src/workers/parser/mapExpression.js]: Why is this needed? Are we trying to prevent unnecessary parsing?

Comment (nchevobbe) [reply src/workers/parser/mapExpression.js]: Yes. we may end up doing 3 mappings, or none, so this should prevent doing multiple ones. Also, we don't want to bail out on the entire mapping since we do want to still map the await expression if it has a syntax error

File: src/workers/parser/mapExpression.js [hunk 2]
```diff
@@ -30,22 +32,23 @@ export default function mapExpression(
     originalExpression: false
   };
 
+  const getAst = memoize(parseConsoleScript).bind(null, expression);
```
Comment (jasonLaster) [inline src/workers/parser/mapExpression.js]: do you need? `memoize(parseConsoleScript).bind(null, expression);` 

I feel like `getAst(expression)` would work just as well

File: src/workers/parser/tests/mapExpression.spec.js [hunk 3]
```diff
@@ -294,6 +301,20 @@ describe("mapExpression", () => {
         originalExpression: false
       }
     },
+    {
+      name: "await (with SyntaxError)",
+      expression: "await new Promise())",
+      newExpression: formatAwait("await new Promise())"),
+      parseExpression: false,
```
Comment (jasonLaster) [inline src/workers/parser/tests/mapExpression.spec.js]: what's this doing?

File: src/workers/parser/tests/mapOriginalExpression.spec.js [hunk 4]
```diff
@@ -3,31 +3,35 @@
...
+import { parseConsoleScript } from "../utils/ast";
 import { format } from "prettier";
 
 const formatOutput = output =>
   format(output, {
     parser: "babylon"
   });
 
+const mapOgExpression = (expression, mappings) =>
+  mapOriginalExpression(expression, parseConsoleScript(expression), mappings);
```
Comment (jasonLaster) [inline src/workers/parser/tests/mapOriginalExpression.spec.js]: maybe it would be better to use the public API `mapExpression`?

File: src/workers/parser/mapExpression.js (hunk 2 above)
Comment (nchevobbe) [reply src/workers/parser/mapExpression.js]: right, this can be moved to another PR

File: src/workers/parser/tests/mapExpression.spec.js (hunk 3 above)
Comment (nchevobbe) [reply src/workers/parser/tests/mapExpression.spec.js]: we don't want to parse the expression like for the other expression, because it would throw (see https://github.com/devtools-html/debugger.html/pull/7695/files/07ef9a76c3f1c5dbed0dfbdd096786dcc15344bd#diff-fff0d7c8c37a491198a0e4f041c09ea5L23) 
So we need to explicitly indicate that we don't want to parse it. And then, we check that an invalid expression (note the extra closing parens) is still wrapped in an async iife.

File: src/workers/parser/mapExpression.js (hunk 2 above)
Comment (nchevobbe) [reply src/workers/parser/mapExpression.js]: I removed the `getAst` function and just put the result in an `ast` variable. We'll handle memoization in a new PR

File: src/workers/parser/tests/mapOriginalExpression.spec.js (hunk 4 above)
Comment (nchevobbe) [reply src/workers/parser/tests/mapOriginalExpression.spec.js]: done
